### PR TITLE
fix(web): Retrieve dependent pipelines correctly

### PR DESCRIPTION
**Pull request by nandak522**
### Issue Summary
This fixes the bug regarding `/v2/pipelineTemplates/<pipeline-template-id>/dependentPipelines` API.
Original Issue: https://github.com/spinnaker/spinnaker/issues/6472

### Environment
Spinnaker: `1.33.0+`
Front50: `us-docker.pkg.dev/spinnaker-community/docker/front50:2.31.0`

### Description
Given a `templateId`, the `getDependentConfigs` method called by `listDependentPipelines` method, was returning `null` for pipeline configuration, causing the entire API to return `[]` even if a specific pipeline template had dependent pipelines.. This seems to be because of 👇 change.
```java
import com.netflix.spinnaker.front50.model.pipeline.TemplateConfiguration;
TemplateConfiguration v1Config = objectMapper.convertValue(templatedPipeline.getConfig(), TemplateConfiguration.class);

// `templatedPipeline.getConfig()` above was returning `null`
```
This 👆 I believe is meant for `v1` templates. But for `v2` templates I believe it has to be 👇 :
```java
import com.netflix.spinnaker.front50.model.pipeline.V2TemplateConfiguration;

V2TemplateConfiguration config = objectMapper.convertValue(templatedPipeline, V2TemplateConfiguration.class);
```

Here are the swagger-ui screenshots showing before and after the fix 👇 
| Before | After |
|--------|--------|
| Before | After|

Here are the cURL responses showing before and after the fix 👇 
#### Before
```sh
curl -X GET "http://localhost:30084/v2/pipelineTemplates/9d92ff72-2bbf-43a3-bd57-23c221387051/dependents" -H "accept: */*"
[]
```

#### After
```sh
curl -X GET "http://localhost:30084/v2/pipelineTemplates/9d92ff72-2bbf-43a3-bd57-23c221387051/dependents" -H "accept: */*"
[{"application":"products","pipelineConfigId":"9aef1986-48cf-44ce-859c-97b3042388db","pipelineName":"[Mother Pipeline] Global Deployment Express"}]
```

### CODE REVIEW

File: front50-web/src/main/java/com/netflix/spinnaker/front50/controllers/V2PipelineTemplateController.java [hunk 1]
```diff
@@ -186,22 +186,23 @@ List<String> getDependentConfigs(String templateId) {
...
-                TemplateConfiguration config =
-                    objectMapper.convertValue(
-                        templatedPipeline.getConfig(), TemplateConfiguration.class);
 
-                source = config.getPipeline().getTemplate().getSource();
+                V2TemplateConfiguration config =
+                    objectMapper.convertValue(templatedPipeline, V2TemplateConfiguration.class);
+                source = config.getTemplate().getReference();
               } catch (Exception e) {
+                e.printStackTrace();
```
Comment (dbyron-sf) [inline front50-web/src/main/java/com/netflix/spinnaker/front50/controllers/V2PipelineTemplateController.java:201]: We don't typically print to standard out in shipping code.  If there's something to log here, please do.

**Comment by dbyron-sf** [comment]
Can you add some automated tests that demonstrate the bug?  I'd also like to see a more specific commit message that mentions what was broken.  Thanks.

File: front50-web/src/main/java/com/netflix/spinnaker/front50/controllers/V2PipelineTemplateController.java (hunk 1 above)
Comment (nandak522) [reply front50-web/src/main/java/com/netflix/spinnaker/front50/controllers/V2PipelineTemplateController.java]: This is removed.

**Comment by nandak522** [comment]
@dbyron-sf I have added the tests. And reworded the commit message. Please take a look.

**Comment by nandak522** [comment]
> Can you add some automated tests that demonstrate the bug? 
Here it is: @dbyron-sf : https://github.com/spinnaker/front50/pull/1459/commits/1402d7b651ba94cc2fe80bb533f861ab9a8629b8#diff-f2c1cc675d8e520ff0541b397bd0d390483bcf197b0576fe2b5976a8db2de02a

**Review by dbyron-sf — APPROVED** [review]

**Pull request merged**